src/Aspire.Hosting.Azure/Provisioning/Internal/DefaultProvisioningContextProvider.cs

```diff
@@ -32,192 +30,190 @@ internal sealed class DefaultProvisioningContextProvider(
-                """
-                The model contains Azure resources that require an Azure Subscription. 
-                Please provide the required Azure settings.
-                <br /><br />
-                If you do not have an Azure subscription, you can create a <a href="https://azure.com/free" target="_blank">free account</a>.
-                """,
-                [
-                    new InteractionInput { InputType = InputType.Select, Label = "Location", Placeholder = "Select Location", Required = true, Options = [..locations] },
-                    new InteractionInput { InputType = InputType.Password, Label = "Subscription ID", Placeholder = "Select Subscription ID", Required = true },
-                ],
-                new InputsDialogInteractionOptions { ShowDismiss = false, EscapeMessageHtml = false },
-                cancellationToken).ConfigureAwait(false);
-
-        _options.Location = result.Data?[0].Value;
-        _options.SubscriptionId = result.Data?[1].Value;
-
-        _channel.Writer.TryWrite(true);
-    }
-
-    private ResourceCommandState ShowConfigCommand(UpdateCommandStateContext context)
-    {
-        if (_options.Location == null || _options.SubscriptionId == null)
-        {
-            return ResourceCommandState.Enabled;
+            var messageBarResult = await interactionService.PromptMessageBarAsync(
+                 "Azure Provisioning",
+                 "The model contains Azure resources that require an Azure Subscription.",
+                 new MessageBarInteractionOptions
+                 {
+                     Intent = MessageIntent.Warning,
+                     PrimaryButtonText = "Enter values"
+                 },
+                 cancellationToken)
+                 .ConfigureAwait(false);
+
+            if (messageBarResult.Data)
+            {
+                var result = await interactionService.PromptInputsAsync(
+                    "Azure Provisioning",
+                    """
+                    The model contains Azure resources that require an Azure Subscription. 
+                    Please provide the required Azure settings.
+
+                    If you do not have an Azure subscription, you can create a [free account](https://azure.com/free).
+                    """,
+                    [
+                        new InteractionInput { InputType = InputType.Choice, Label = "Location", Placeholder = "Select Location", Required = true, Options = [..locations] },
+                        new InteractionInput { InputType = InputType.SecretText, Label = "Subscription ID", Placeholder = "Select Subscription ID", Required = true },
+                        new InteractionInput { InputType = InputType.Text, Label = "Resource Group", Value = "GetResourceGroupDefault"},
+                    ],
+                    new InputsDialogInteractionOptions { ShowDismiss = false, EnableMessageMarkdown = true },
```

Comment:
It should be possible to dismiss, allowing users to interact with the dashboard and other resources still.

Reply:
It is possible.

Reply:
Can you also query for subscription ids?

Reply:
I can, but doing that would prevent me from entering in something that isn't in the list. I plan on openining an interaction service feature request tomorrow to support an input that allows both a list of options and the ability to enter in a free form value.

Without this, do you think for 9.4 we should leave it as-is without the list? Or should we query the for the list and show it if we found some (preventing entering something that isn't in the list)?

Reply:
I’m fine with it as is for 9.4. Open an issue to improve it in 9.5 to query for subs.

Reply:
You've set `ShowDismiss = false` but the dismiss button (top right cross) is still visible? That would be a bug.

Reply:
Are values remembered? A checkbox was added to the parameters dialog for that.

Reply:
@eerhardt yes we need to save them to user secrets